### PR TITLE
Fix function signature mismatch issue in RISC-V build

### PR DESCRIPTION
**Pull request by ksco**
Fixes DynamoRIO compilation issue on a RISC-V SBC (LicheePi 4A).

Changes the signature of `get_special_ibl_linkstub` declaration to match its definition.

Other compilation issues are fixed in PR #6058, after which it compiles normally.

### CODE REVIEW

File: core/link.h [hunk 1]
```diff
@@ -511,7 +511,7 @@ get_hot_patch_linkstub(void);
 const linkstub_t *
 get_client_linkstub(void);
 const linkstub_t *
-get_special_ibl_linkstub(int index, bool is_trace);
+get_special_ibl_linkstub(ibl_branch_type_t ibl_type, bool is_trace);
```
Comment (abhinav92003) [inline core/link.h:514]: nit about the PR description and title: see the example string at  https://dynamorio.org/page_code_reviews.html.

For the description, please use something like the following:
```
Fixes DynamoRIO compilation issue on a RISC-V SBC. 

Changes the signature of get_special_ibl_linkstub declaration to match its definition.

Other compilation issues are fixed in PR #6058, after which it compiles normally.
```

and the title should provide a summary too: `Fix function signature mismatch issue in RISC-V build`

**Review by abhinav92003 — APPROVED** [review]

File: core/link.h (hunk 1 above)
Comment (ksco) [reply core/link.h:514]: Wow, thanks for the detailed guidance! I've updated the title and desc as suggested.

**Comment by abhinav92003** [comment]
Since you're not in our Committers list yet, I can merge his PR for you.

**Comment by ksco** [comment]
@abhinav92003 CI is finished, please merge this, thanks!

**Comment by abhinav92003** [comment]
> @abhinav92003 CI is finished, please merge this, thanks!

Since the merge checks were still running, I decided to merge it the next day. Doing it now.

**Comment by ksco** [comment]
Thanks, feel free to do it anytime. ;)

**Pull request merged**